src/plugin.js

```diff
@@ -85,6 +86,19 @@ const initPlugin = function(player, options) {
 
   // clears the previous monitor timeout and sets up a new one
   const resetMonitor = function() {
+    // start the loading spinner if player has stalled
+    window.clearTimeout(waiting);
+    player.removeClass('vjs-waiting');
+    waiting = window.setTimeout(function() {
```

Comment:
We should be using `player.setTimeout` across the board to prevent memory leaks.